fms/utils/generation.py

```diff
@@ -53,17 +135,35 @@ def generate(
         use_cache: requires that the model accept use_cache and
             past_key_value_states args in forward method.
     """
-    batched = False
     if num_beams != 1:
         raise NotImplementedError("generate() does yet not support beam search")
-    if type(input_ids) == torch.Tensor:
-        if input_ids.dim() != 1:
-            batched = True
-    else:
-        raise RuntimeError("generate() requires a tensor of token ids as the prefix")
 
-    if not batched:
+    if isinstance(input_ids, torch.Tensor) and len(input_ids.shape) == 1:
+        model_input_lengths = [input_ids.size(0)]
         input_ids = input_ids.unsqueeze(0)
+        is_batch = False
+        position_ids = None
+    else:
+        bsize = len(input_ids)
+        if isinstance(input_ids, torch.Tensor):
+            max_len = input_ids.size(1)
+            model_input_lengths = [max_len for _ in range(bsize)]
+            position_ids = None
+        else:
+            max_len = max([seq.size(0) for seq in input_ids])
+            model_input_lengths = [seq.size(0) for seq in input_ids]
+            # Setting this to 0, however if 0 is the eos, we will end up truncating the output if using truncate_after_eos
+            # once this workflow works for nested tensor, this can probably be removed
+            input_ids = torch.stack(
+                [
+                    F.pad(input_ids[i], (max_len - model_input_lengths[i], 0))
+                    for i in range(bsize)
+                ]
+            )
+            position_ids = __create_prefill_position_ids(
+                model_input_lengths, input_ids.device
+            )
+        is_batch = True
```

Comment:
if this was a list of length 1 then it's not actually batched right?

Reply:
ah true, I think it should be false inn that case